addons/account_edi/models/account_edi_format.py

```diff
@@ -398,7 +398,7 @@ def _decode_attachment(self, attachment):
 
         return to_process
 
-    def _create_document_from_attachment(self, attachment):
+    def _create_document_from_attachment(self, attachment, journal=None):
```

Comment:
how is it not related to `self.env['account.move']._create_document_from_attachment()` ? I thought this PR aimed to solve that...

notice the 2 functions have the same name but different args :face_with_spiral_eyes:

Reply:
I meant `self.env['account.journal']._create_document_from_attachment()`

Reply:
https://github.com/odoo-dev/odoo/blob/6d7fd5649d92b2097d24c07a7012f6f2c1450705/addons/account/models/account_journal.py#L707
Here, we are in an `account.journal` which searches for decoders here. Whenever it finds one, it calls it with the attachment and self (the journal in which we want to upload the invoice). 

`AccountEdiFormat._create_document_from_attachment(self, attachment, journal=None)` is a decoder (here https://github.com/odoo/odoo/pull/101280/files/6d7fd5649d92b2097d24c07a7012f6f2c1450705#diff-27ab736c17e6fa2dfe347456b3c128d29e7401fdcc265ebaf92abe74019dcf3fR401) which accepts an attachment and a journal.